execution_engine/src/core/engine_state/mod.rs

```
@@ -99,6 +103,16 @@ pub struct EngineState<S> {
     state: S,
 }
 
+impl EngineState<LmdbGlobalState> {
+    pub fn flush_environment(&self) {
+        self.state
+            .environment
+            .env()
+            .sync(true)
+            .expect("should flush environment")
```

Let's just propagate the error here, rather than expect.